core/src/main/java/org/apache/carbondata/core/datastore/columnar/BlockIndexerStorageForShort.java

```diff
@@ -192,12 +192,23 @@ private void compressDataMyOwnWay(ColumnWithShortIndex[] indexes) {
     }
     map.add(start);
     map.add(counter);
-    this.keyBlock = convertToKeyArray(list);
```

Comment:
This is a comment for `compressMyOwnWay` function, suggest to use `indexes.length / 2` as the initial size to allocate the ArrayList, instead of 10, which is too small and will cause repeated arraylist expansion